modules/services/spotifyd.nix

```
@@ -0,0 +1,159 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.spotifyd;
+
+  configFile = pkgs.writeText "spotifyd.conf" ''
+    [global]
+    username = ${cfg.user}
+    ${optionalString (cfg.password != null) "password = ${cfg.password}"}
+    ${optionalString (cfg.password_cmd != null) "password_cmd = \"${cfg.password_cmd}\""}
+    ${optionalString cfg.keyring "use_keyring = true"}
+    backend = ${cfg.backend}
+    device = ${cfg.device}
+    ${optionalString (cfg.control_device != null) "control = ${cfg.control_device}"}
+    mixer = ${cfg.mixer}
+    volume_controller = ${cfg.alsa_audio_device}
+    ${optionalString (cfg.song_change_hook != null) "on_song_change_hook = \"${cfg.song_change_hook}\""}
+    device_name = ${cfg.name}
+    bitrate = ${toString cfg.bitrate}
+    cache_path = ${cfg.cache}
+    no_audio_cache = ${toString cfg.audio_cache}
+    volume_normalisation = ${toString cfg.normalize}
+    normalisation_pregain = ${toString cfg.pregain}
+    ${optionalString (cfg.zeroconf != null) "zeroconf_port = ${toString cfg.zeroconf}"}
+  '';
+in {
+  options.services.spotifyd = {
+    enable = mkEnableOption "SpotifyD connect";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.spotifyd;
+      description = "The spotifyd package to use.";
+    };
+
+    user = mkOption {
+        type = types.str;
+        description = ''
+          Spotify username to authentificate (must be premium)
+        '';
+    };
+
+    password = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "Spotify password";
+    };
+
+    password_cmd = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "use command to pipe out the spotify password";
+    };
+
+    keyring = mkOption {
+        type = types.bool;
+        default = false;
+        description = "use keyring to get spotify password";
```

Option descriptions should be at least a full sentences, i.e., start with capital letter and end in a period. For this option, for example, it would require changing
```suggestion
        description = "Use keyring to get spotify password.";
```

---

remove that option completly